File: apps/control-plane/api/router/kernel_upgrade.go
package router

import (
	"github.com/gin-gonic/gin"
	"trojan-panel/api"
)

func initKernelUpgradeRouter(trojanApi *gin.RouterGroup) {
	kernel := trojanApi.Group("/kernel")
	{
		kernel.GET("/releases", api.KernelReleases)
		kernel.GET("/inventory", api.KernelInventory)
		kernel.POST("/createTask", api.CreateKernelTask)
		kernel.GET("/selectTaskPage", api.SelectKernelTaskPage)
		kernel.GET("/selectTaskById", api.SelectKernelTaskById)
		kernel.POST("/retryTask", api.RetryKernelTask)
		kernel.POST("/probeMTLS", api.ProbeKernelMTLS)
	}
}

File: apps/control-plane/web/src/api/kernel-upgrade.js
import request from '@/utils/request'

export function kernelReleases(params) {
  return request({ url: '/kernel/releases', method: 'get', params })
}

export function kernelInventory(params) {
  return request({ url: '/kernel/inventory', method: 'get', params })
}

export function createKernelTask(data) {
  return request({ url: '/kernel/createTask', method: 'post', data })
}

export function selectKernelTaskPage(params) {
  return request({ url: '/kernel/selectTaskPage', method: 'get', params })
}

export function selectKernelTaskById(params) {
  return request({ url: '/kernel/selectTaskById', method: 'get', params })
}

export function retryKernelTask(data) {
  return request({ url: '/kernel/retryTask', method: 'post', data })
}

export function probeKernelMTLS(data) {
  return request({ url: '/kernel/probeMTLS', method: 'post', data })
}
